assets/src/components/ui/entity-list/entity-list.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * External imports
+ */
+import { without } from 'lodash';
+import { __ } from '@eventespresso/i18n';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import './style.css';
+import withFormContainerAndPlaceholder
+	from '../../form/base/with-form-container-and-placeholder';
+
+/**
+ * EntityList
+ * base component for displaying a list of entities (dates, tickets, etc)
+ * as either a list table or grid of entity blocks
+ *
+ * @function
+ * @param {Array} entities
+ * @param {Component} EntityGridView
+ * @param {Component} EntityListView
+ * @param {string} htmlClass
+ * @param {string} view
+ * @param {string} noResultsText
+ * @param {mixed} otherProps
+ * @return {Component} list of rendered entities
+ */
+const EntityList = ( {
+	entities,
+	EntityGridView,
+	EntityListView,
+	htmlClass,
+	view,
+	noResultsText,
+	...otherProps
+} ) => {
+	// verify array and remove undefined
+	entities = Array.isArray( entities ) ?
```

Comment:
React is using unidirectional data flow also called one-way binding. Mutating prop is a form of two-way data binding which might create some unwanted results in context of react.

Reply:
@nicolad thank you for pointing that out. As React docs also mention
>**All React components must act like pure functions with respect to their props.**

Reply:
Yeah, especially with the introduction of hooks, they went all in with FP concepts.

Reply:
:facepalm: doh

ya thanks @nicolad, I'm well aware that props should never be mutated, but obviously forget from time to time.

Reply:
@tn3rb I also do forget some subtle details and because of that I would rather use a ESLint rule like this: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prefer-read-only-props.md
rather than worrying about all the rules and just focus on the problem at hand.